cmd/influxd/run/startup_logger.go

```diff
@@ -0,0 +1,86 @@
+package run
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"go.uber.org/zap"
+)
+
+// TODO: Add metrics for in memory indexes and TSI indexes
+type StartupProgressLogger struct {
+	tsmFilesCompleted     atomic.Uint64
+	tsmFileBytesCompleted atomic.Uint64
+	tsmFilesTotal         atomic.Uint64
+	tsmFileBytesTotal     atomic.Uint64
+	shardsCompleted       atomic.Uint64
+	shardsTotal           atomic.Uint64
+	walFilesCompleted     atomic.Uint64
+	walFileBytesCompleted atomic.Uint64
+	walFilesTotal         atomic.Uint64
+	walFileBytesTotal     atomic.Uint64
+	logger                *zap.Logger
+}
+
+func NewStartupProgressLogger(logger *zap.Logger) *StartupProgressLogger {
+	return &StartupProgressLogger{
+		logger: logger,
+	}
+}
+
+func (s *StartupProgressLogger) AddTSMFile() {
+	s.tsmFilesTotal.Add(1)
+}
+
+func (s *StartupProgressLogger) AddTSMFileBytes(size uint64) {
+	s.tsmFileBytesTotal.Add(size)
+}
+
+func (s *StartupProgressLogger) CompletedTSMFile(size uint64) {
+	filesCompleted := s.tsmFilesCompleted.Add(1)
+	bytesCompleted := s.tsmFileBytesCompleted.Add(size)
+	totalFiles := s.tsmFilesTotal.Load()
+	totalBytes := s.tsmFileBytesTotal.Load()
+
+	percentFiles := float64(filesCompleted) / float64(totalFiles) * 100
+	percentBytes := float64(bytesCompleted) / float64(totalBytes) * 100
+	s.logger.Info(fmt.Sprintf("Finished loading TSM file, current progress %.1f%% files (%d / %d), %.1f%% bytes (%d / %d).", percentFiles, filesCompleted, totalFiles, percentBytes, bytesCompleted, totalBytes))
+}
```

Comment:
Since it sounds like you want to add the usage of these methods to follow-on PRs, and they aren't called yet, these shouldn't be in this PR. Put them in the PR where they get used.

Reply:
Sounds good will do 👍